llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp

```diff
@@ -95,7 +95,7 @@ using namespace llvm;
 static cl::opt<bool>
     LowerCtorDtor("nvptx-lower-global-ctor-dtor",
                   cl::desc("Lower GPU ctor / dtors to globals on the device."),
-                  cl::init(false), cl::Hidden);
+                  cl::init(true), cl::Hidden);
```

Comment:
This will allow global constructors/destructors for all users by default, but it relies on something to call those constructors and that something will only be provided by OpenMP. We do want to diagnose it if there's no runtime support avaialble.

I think we still need to keep false as the default.

Reply:
This was the easiest way to get the desired effect. Passing `--nvptx-lower-glocal-ctor-dtor` is subtly broken because I think it will fail if the user didn't build with the NVPTX target. The OpenMP runtime is supposed to be buildable for NVPTX even without backend support so I was worried it would degrade that. Do you think I could check for `openmp` module data and set it based off of that? OpenMP always emits an `openmp` global flag.

Reply:
Allowing ctor generation based on module metadata may indeed be a reasonable way to deal with it.

Reply:
Done